***************
*** 3,10 ****
--- 3,38 ----
  import django.core.validators
  from django.db import migrations, models
  import manager.validators
+ import numpy
+ from django.core.files.base import ContentFile
+ from manager.glb_top_view import generateOrthoView
+ from PIL import Image
+ import os
+ import pprint

+ os.environ["EGL_PLATFORM"] = "surfaceless"

+ def preserve_map3d(apps, schema_editor):
+   Scene = apps.get_model('sscape', 'Scene')
+   for scene in Scene.objects.all():
+     if scene.map_3d:
+       scene.map=scene.map_3d
+     scene.save()
+
+ def generate_and_save_thumbnail(apps, schema_editor):
+   Scene = apps.get_model('sscape', 'Scene')
+   for scene in Scene.objects.all():
+     if scene.map and os.path.splitext(scene.map.path)[1].lower() == ".glb":
+         ext = os.path.splitext(scene.map.path)[1].lower()
+         if ext == ".glb":
+             img_data, pixels_per_meter = generateOrthoView(scene, scene.map.path)
+             scene.scale = pixels_per_meter
+             img = Image.fromarray(numpy.uint8(img_data))
+             with ContentFile(b'') as imgfile:
+                 img.save(imgfile, format='PNG')
+                 scene.thumbnail.save(scene.name + '_2d.png',imgfile,save=False)
+             scene.save()
+
  class Migration(migrations.Migration):

      dependencies = [
***************
*** 12,17 ****
--- 40,46 ----
      ]

      operations = [
+         migrations.RunPython(preserve_map3d),
          migrations.RemoveField(
              model_name='scene',
              name='map_3d',
***************
*** 24,29 ****
--- 53,59 ----
              model_name='scene',
              name='map_alt2',
          ),
+         migrations.RunPython(generate_and_save_thumbnail),
          migrations.AlterField(
              model_name='scene',
              name='map',
