***************
*** 6,11 ****
--- 6,21 ----
  from django.conf import settings
  from django.db import migrations, models

+ def transfer_transforms_data(apps, schema_editor):
+     Cam = apps.get_model('manager', 'Cam')
+     for cam in Cam.objects.all():
+         cam.transforms = [
+             cam.transform1, cam.transform2, cam.transform3, cam.transform4,
+             cam.transform5, cam.transform6, cam.transform7, cam.transform8,
+             cam.transform9, cam.transform10, cam.transform11, cam.transform12,
+             cam.transform13, cam.transform14, cam.transform15, cam.transform16
+         ]
+         cam.save()

  class Migration(migrations.Migration):

***************
*** 22,27 ****
--- 32,43 ----
                  ('is_ready', models.BooleanField(default=False)),
              ],
          ),
+         migrations.AddField(
+             model_name='cam',
+             name='transforms',
+             field=django.contrib.postgres.fields.ArrayField(base_field=models.FloatField(blank=True, default=None, null=True), blank=True, default=list, size=None),
+         ),
+         migrations.RunPython(transfer_transforms_data),
          migrations.RemoveField(
              model_name='cam',
              name='transform1',
***************
*** 127,137 ****
              name='threshold',
              field=models.FloatField(blank=True, default=None, null=True),
          ),
!         migrations.AddField(
!             model_name='cam',
!             name='transforms',
!             field=django.contrib.postgres.fields.ArrayField(base_field=models.FloatField(blank=True, default=None, null=True), blank=True, default=list, size=None),
!         ),
          migrations.AddField(
              model_name='childscene',
              name='retrack',
--- 143,149 ----
              name='threshold',
              field=models.FloatField(blank=True, default=None, null=True),
          ),
!
          migrations.AddField(
              model_name='childscene',
              name='retrack',
