***************
*** 28,97 ****
              model_name='childscene',
              name='sscape_childscene_unique_relationships',
          ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_x1',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_x2',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_x3',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_x4',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_y1',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_y2',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_y3',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='cam_y4',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_x1',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_x2',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_x3',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_x4',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_y1',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_y2',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_y3',
!         ),
!         migrations.RemoveField(
!             model_name='cam',
!             name='map_y4',
!         ),
          migrations.RemoveField(
              model_name='scene',
              name='map_3d',
--- 28,113 ----
              model_name='childscene',
              name='sscape_childscene_unique_relationships',
          ),
!         migrations.RenameField(
!               model_name='cam',
!               old_name='cam_x1',
!               new_name='transform1'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_y1',
!               new_name='transform2'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_x2',
!               new_name='transform3'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_y2',
!               new_name='transform4'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_x3',
!               new_name='transform5'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_y3',
!               new_name='transform6'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_x4',
!               new_name='transform7'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='cam_y4',
!               new_name='transform8'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_x1',
!               new_name='transform9'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_y1',
!               new_name='transform10'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_x2',
!               new_name='transform11'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_y2',
!               new_name='transform12'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_x3',
!               new_name='transform13'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_y3',
!               new_name='transform14'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_x4',
!               new_name='transform15'
!           ),
!           migrations.RenameField(
!               model_name='cam',
!               old_name='map_y4',
!               new_name='transform16'
!           ),
          migrations.RemoveField(
              model_name='scene',
              name='map_3d',
***************
*** 131,216 ****
          ),
          migrations.AddField(
              model_name='cam',
-             name='transform1',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform10',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform11',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform12',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform13',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform14',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform15',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform16',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform2',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform3',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform4',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform5',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform6',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform7',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform8',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
-             name='transform9',
-             field=models.FloatField(blank=True, default=None, null=True),
-         ),
-         migrations.AddField(
-             model_name='cam',
              name='transform_type',
              field=models.CharField(choices=[('matrix', 'Matrix'), ('euler', 'Euler Angles'), ('quaternion', 'Quaternion'), ('homography', 'Homography')], default='homography', max_length=10),
          ),
--- 147,152 ----
